Filter results

jerrylin03 opened the pull request:
Filter fetched real estate property results by minimum and maximum price ranges from user profile

jerrylin03 committed:
feat: Add price to rent ratio

-grab data from multiple JSON
-add id argument to parser
-add score function to parser

jerrylin03 committed:
feat: add filter to property display

- filter properties based on min & max price range from profile
- calculate monthly mortgage payments based on US mortgage rates
- pretty output for filter

jlin332 commented on scanner_cli.py:
```diff
@@ -245,23 +301,93 @@ def store_user_data(username):
         store_user_data(args.username)
     elif args.function == "fetch":
         parse_external_data()
+    elif args.function == "parse":
```
what is the difference from fetch/parse? same function getting called

jlin332 commented on scanner_cli.py:
```diff
@@ -245,23 +301,93 @@ def store_user_data(username):
+
+    # Find the median price based on state and city
+    median_price = None
+    for state_data in median_price_data:
```
is this median_price_data a dict? or array? Look into using https://docs.python.org/3/library/typing.html

jlin332 commented on scanner_cli.py:
```diff
@@ -205,21 +260,22 @@ def store_user_data(username):
             update = input("Do you want to update the user profile ? (Y/N) ")
         else:
             print(f"New user {username}")
+            update = "Y"
```
what is the update doing? why Y?

jerrylin03 replied:
update variable takes in user input after displaying current profile filters so we have the option to reinput fields

jerrylin03 committed:
fix: revise PR comments

- remove redundant iteration in median data dictionary
- remove duplicate function argument for parsing external data
